frontend/discourse/app/components/topic-list/item.gjs

```diff
@@ -60,7 +60,10 @@ export default class Item extends Component {
   }
 
   get tagClassNames() {
-    return this.args.topic.tags?.map((tagName) => `tag-${tagName}`);
+    return this.args.topic.tags?.map((tag) => {
+      const tagName = typeof tag === "string" ? tag : tag.name;
```

Comment:
Locally, I have this prepended in all the frontend files affected

```
      if (typeof tag === "string") {
        // print the trace here so we know who is violating
        console.warn(
          `Topic tag is a string (${tag}) - this will be deprecated soon. Please pass tag objects instead.`
        );
        console.warn(new Error().stack);
      }
```

But am thinking of a better way.